### PR TITLE
feat: implement mjai replay and demo ML project

### PR DESCRIPTION
**Pull request by smly**
Resolves https://github.com/smly/RiichiEnv/issues/75

This PR implements MJAI log loading functionality and enhances feature extraction(`grp_features`) for Global Reward Predictor (GRP) training (`ml-baseline`).

### CODE REVIEW

File: demos/ml_baseline/train_grp.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+Usage:
+    python train_grp.py
+"""
+import tqdm
+import polars as pl
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
```
[the remark this answers is not in the document]
Comment (smly) [reply demos/ml_baseline/train_grp.py]: Resolved in https://github.com/smly/RiichiEnv/pull/76/commits/f3a2dd770d3fc72caa0e88e50e9a5098d4661f01

File: demos/ml_baseline/train_grp.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        "ju",
+        "ben",
+        "liqibang",
+    ]).to_dicts()
+
+    point_weights = [100, 40, -40, -100]
+    player_idx = 0
+
+    device_str = "cuda"
+    device = torch.device(device_str)
```
[the remark this answers is not in the document]
Comment (smly) [reply demos/ml_baseline/train_grp.py]: Resolved in https://github.com/smly/RiichiEnv/pull/76/commits/f3a2dd770d3fc72caa0e88e50e9a5098d4661f01
